scripts/pre_commit_linter.py

```diff
@@ -1463,6 +1471,107 @@ def _check_directive_scope(all_files):
     return summary_messages
 
 
+def _check_sorted_dependencies(all_files, parsed_js_files):
+    """This function checks that the dependencies which are
+    imported in the controllers/directives/factories in JS
+    files are in following pattern: dollar imports ,regular
+    imports and constant imports, all in sorted order.
+    """
+    print 'Starting sorted dependencies check'
+    print '----------------------------------------'
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
```

Comment:
This is a bit hard to read, please make `filename.endswith('.js')` the first condition checked to improve!

Reply:
Done